testing/test_parseopt.py

```diff
@@ -298,7 +298,11 @@ def test_argcomplete(testdir, monkeypatch):
         # redirect output from argcomplete to stdin and stderr is not trivial
         # http://stackoverflow.com/q/12589419/1307905
         # so we use bash
-        fp.write('COMP_WORDBREAKS="$COMP_WORDBREAKS" python -m pytest 8>&1 9>&2')
+        fp.write(
+            'COMP_WORDBREAKS="$COMP_WORDBREAKS" {} -m pytest 8>&1 9>&2'.format(
+                sys.executable
```

Comment:
should ~probably be `shlex.quote(sys.executable)` since we're sticking it into a shell script

Reply:
yep!

Reply:
`{!r}` might be enough though.

Reply:
repr is not a substitute for proper shell quoting, that's asking for shell injection ;)

Reply:
Sure.. but this is something we have control over here more or less.
It's good to use the proper way though, of course.. :)

Reply:
we don't control this at all though, it's a user's directory -- they could do all sorts of wild things with the directory they clone/install pytest into

Reply:
I've meant the user / test runner - who is setting/using `sys.executable`.

Reply:
of course, that doesn't stop people from having spaces in their development dirs, or non-ascii usernames, or having quotes in their names

Reply:
Spaces would be handled by the `repr()` then.
Anyway, your fix is better.